### ISSUE TITLE
Chrome Canary version "blocks" the atmospherejs search site

### ISSUE THREAD

**Issue by mcrayne** [percolatestudio/atmosphere#433]
This is what you get at the normal URL:

![image](https://cloud.githubusercontent.com/assets/3066333/13825238/08ed9c9e-eb88-11e5-8de4-4915aa555bc9.png)

Works fine in non-Canary Chrome. I am experiencing this issue with my apps also.

**Comment by tmeasday** [comment]
Any idea what the issue is?

**Comment by mcrayne** [comment]
Thanks for getting back to me. No, except that it appears Canary is actually flagging files I download from the google site as being untrustworthy. I deleted the Canary browser and am using Chrome and IE (my clients all use IE, I hate it but what can you do right?). I just thought it was odd that your site and my puny project suffered similar issues. 

I am going to re-install a fresh copy of Canary and see if it makes a difference.

Mike

**Comment by zol** [comment]
Thanks Mike. Let us know if newer versions of Canary fix this. I suspect/hope it's their bug.

**Comment by tmeasday** [comment]
Yeah given it's hitting our two sites, there's a fair chance it's their bug and not ours ;)

**Comment by mcrayne** [comment]
I reloaded Canary and still have the problem. I guess I wont be using Canary for a while. I wish I had time to troubleshoot it but I am having my own problems now trying to load an app to Galaxy. Not looking good – I deploy from the CLI and it is successful but when I go to the URL it is not found because there is no application registered. The MDG is no help at all with their online stuff – just makes me feel stupid when I try to follow their instructions.

Thanks for getting back to me.

Mike

**Comment by tmeasday** [comment]
> The MDG is no help at all with their online stuff – just makes me feel stupid when I try to follow their instructions.

That's no good; which instructions are you finding hard to follow?

**Comment by mcrayne** [comment]
Deploying an app to meteorapp.com – which I assume is galaxy. Going to galaxy.meteor.com is just the marketing site. It seems to work basically the same as the free hosting did with a different root URL. When I deploy from my CLI it says success. But when I go to the URL I get:  

404 Not Found: No applications registered for host 'imua.meteorapp.com'.

Not much help. Then finding the help button is torture because in “galaxy” the help button in the upper right takes you to the meteor page. I could go on but it is a waste of your time. The weirdest instructions they have is this:

Specify this host name during meteor deploy:

DEPLOY_HOSTNAME=galaxy.meteor.com meteor deploy <custom.subdomain>.meteorapp.com --settings path-to-settings.json

What the hell is “during meteor deploy”? You obviously can’t just type that instruction into the CLI (I tried like a dumbass) and typing it in after 

> meteor deploy

is stupid because there is “meteor deploy” in the arguments. I figure it is a Mac thing or maybe some environmental variables that are supposed to get set but there is nothing in the knowledge base.

Mike

**Comment by tmeasday** [comment]
Hey @mcrayne I think those instructions need to be better for windows users. That is intended to be a command that you run instead of your regular `meteor deploy` (so at the moment you are still deploying to the old free hosting).

Maybe try:

```
SET DEPLOY_HOSTNAME=galaxy.meteor.com
meteor deploy imua.meteorapp.com --settings path-to-settings.json
```

Let me know if that works

**Comment by mcrayne** [comment]
Wow, that helped, a little. The app deployed and I am using a servo (or whatever they call the container) but I get this:

502 Bad Gateway: Registered endpoints failed to handle the request.

So I got that going for me. I update MDG. Maybe there is some setting I need. This part of the app is just the prototype of the front end, not using any MongoDB yet and it is not in my package list – just like they say. So that isn’t the problem, that I know of. But at least I am getting closer. Thanks Tom.

Mike

**Comment by tmeasday** [comment]
It may be the container is still starting up -- it typically takes a minute or two. Have you checked at https://galaxy.meteor.com?

**Comment by mcrayne** [comment]
Spoke too soon. It wants my mongodb url. Dooh!. But I think you fixed it for me. I can at least see some server logs and know whats up.

Mike

**Comment by tmeasday** [comment]
Yep! I think you can remove the dependency on Mongo if you mess with your `.meteor/packages` (obviously removing `mongo`, but possibly a few other packages that depend on Mongo). 

Otherwise you can get a free db from mLab.

**Comment by mcrayne** [comment]
Yeah, you fixed it. I am going to try compose.io for mongo. I have been using MongoLab with Heroku and that works fine. Right now it is kicking out because it can’t load accounts-base which I assume is trying to create collections in mongo though it is not an option I have of removing the package that I can see. 

Thanks a lot Tom.

**Comment by mcrayne** [comment]
Still not sure what is going on with accessing atmosphere using Canary but here is a screen shot showing the network connection being refused. Really strange that Chrome and IE don’t care.

**Comment by denisbabineau** [comment]
Ran into the same rendering issue with atmosphere site using Canary on OSX; oddly enough I don't see any network connections being refused in my case though. However I do see this error in the console which isn't present with regular Chrome:

```
Exception in callback of async function: Error: Handler with name 'u' already exists.
    at o._create (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:1611)
    at o.push (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:1795)
    at https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:2045
    at Array.forEach (native)
    at Function.A.each.A.forEach (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:1:865)
    at o.append (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:1975)
    at https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:2164
    at Array.forEach (native)
    at Function.A.each.A.forEach (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:1:865)
    at o.append (https://atmospherejs.com/f82ef052a4ae43a3590d5adfc9fac7fdf9dd4e0e.js?meteor_js_resource=true:67:1975)
```

When I break on the line that throws the exception (before it's raised) the callstack looks like this:

```
e._debug (f82ef052a4ae43a…formatted:1266)
e._debug (f82ef052a4ae43a…ormatted:32794)
o (f82ef052a4ae43a…formatted:1319)
e.bindEnvironment (f82ef052a4ae43a…formatted:1328)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13570)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
p (f82ef052a4ae43a…ormatted:13590)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
l (f82ef052a4ae43a…ormatted:13596)
(anonymous function) (f82ef052a4ae43a…ormatted:20454)
t.Computation._compute (f82ef052a4ae43a…formatted:1517)
t.Computation (f82ef052a4ae43a…formatted:1455)
t.autorun (f82ef052a4ae43a…formatted:1634)
(anonymous function) (f82ef052a4ae43a…ormatted:20452)
t.nonreactive (f82ef052a4ae43a…formatted:1645)
d.dispatch (f82ef052a4ae43a…ormatted:20451)
m.dispatch (f82ef052a4ae43a…ormatted:21050)
r (f82ef052a4ae43a…ormatted:21098)
t.Computation._compute (f82ef052a4ae43a…formatted:1517)
t.Computation (f82ef052a4ae43a…formatted:1455)
t.autorun (f82ef052a4ae43a…formatted:1634)
m.start (f82ef052a4ae43a…ormatted:21096)
(anonymous function) (f82ef052a4ae43a…ormatted:20716)
withValue (f82ef052a4ae43a…formatted:1307)
n (f82ef052a4ae43a…formatted:1092)
e.bindEnvironment (f82ef052a4ae43a…formatted:1326)
e (f82ef052a4ae43a…formatted:1042)
postMessage (async)
u ()
defer ()
(anonymous function) ()
o ()
o ()
```

Anyhow, for myself I'll just use regular chrome in the meantime.

**Comment by DanielDornhardt** [comment]
+1 I just found this and I have the same problem using chrome canary. Other meteor projects of mine seem to work fine.

**Comment by tmeasday** [comment]
This was due to https://github.com/iron-meteor/iron-router/issues/1513

Just deployed a fix, thanks for the reports all.
